lib/auth/auth.go

```diff
@@ -844,6 +844,10 @@ type GenerateTokenRequest struct {
 	Roles teleport.Roles `json:"roles"`
 	// TTL is a time to live for token
 	TTL time.Duration `json:"ttl"`
+	// Labels sets token labels, e.g. {env: prod, region: us-west}.
+	// Labels are later passed to resources that are joining
+	// e.g. remote clusters and in the future versions, nodes and proxies.
```

Comment:
Yes, exactly. Nodes and proxies should inherit labels from tokens encoded in the X509 metada that they can't change. This will allow scenarios when nodes are deployed to other tenants. It is less relevant for proxies.